### ISSUE TITLE
MEN010 leads to ArgumentNullException

### ISSUE THREAD

**Issue by millo-k** [menees/Analyzers#1]
The statement:
```cs
System.Threading.Thread.Sleep(300);
```
leads to the following **AD0001 warning**:
```cs
Severity	Code	Description	Project	File	Line	Category	Suppression State	Tool
Warning	AD0001	Analyzer 'Menees.Analyzers.Men010AvoidMagicNumbers' threw an exception of type 'System.ArgumentNullException' with message 'Value cannot be null.
Parameter name: source'.
Exception occurred with following context:
Compilation: Menees.Mwe
SyntaxTree: D:\Menees.Mwe\Mwe\ExampleClass.cs

System.ArgumentNullException: Value cannot be null.
Parameter name: source
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
   at Menees.Analyzers.Settings.IsAllowedNumericLiteralCaller(String callerName)
   at Menees.Analyzers.Men010AvoidMagicNumbers.IsAllowedInvocation(LiteralExpressionSyntax literalExpression, Settings settings)
   at Menees.Analyzers.Men010AvoidMagicNumbers.<>c__DisplayClass9_0.<InAllowedDeclarationContext>b__0(SyntaxNode ancestor)
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
   at Menees.Analyzers.Men010AvoidMagicNumbers.<HandleSyntaxTree>b__17_1(LiteralExpressionSyntax literal)
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at Menees.Analyzers.Men010AvoidMagicNumbers.HandleSyntaxTree(SyntaxTreeAnalysisContext context)
   at Menees.Analyzers.Rules.<>c__DisplayClass13_0.<RegisterSyntaxTreeActionHonorExclusions>b__0(SyntaxTreeAnalysisContext c)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c.<ExecuteSyntaxTreeActionsCore>b__59_1(ValueTuple`2 data)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info)
-----

Suppress the following diagnostics to disable this analyzer: MEN010	Mwe		1	Compiler	Active	Menees.Analyzers
```

**Workaround**:
Adding the `CallerRegex` element to the _Menees.Analyzers.Settings.xml_ file solves the problem:
```cs
  <AllowedNumericLiterals>
    <CallerRegex>^From(Days|Hours|Minutes|Seconds)$</CallerRegex>
  </AllowedNumericLiterals>
```

**Project details**:
- Microsoft Visual Studio Professional: Version 16.8.3
- Target framework: .NET Framework 4.8
- OutputType: Windows Application
- Menees.Analyzers: Version 2.0.5

**Comment by menees** [comment]
Thanks for reporting this and for including detailed repro steps! I've checked in the fix and published it in the 2.0.6 release.

**Comment by millo-k** [comment]
The issue is solved with version 2.0.6.
